src/Registry.ts

```diff
@@ -32,11 +32,11 @@ export default class Registry<T> {
 	 * @returns the matching value, or a default value if one exists.
 	 */
 	match(...args: any[]): T {
-		let entries = this._entries.slice(0);
+		const entries = this._entries ? this._entries.slice(0) : [];
```

Comment:
Occurrences like this are probably something that we should review across the entire code base as a separate concern.

I can of course fix this single one :)